Port can be zero for Dispersy candidate, resulting in AssertionError

See the following error we received in the crash reporter:


```
Traceback (most recent call last):
File "TriblerGUI\event_request_manager.py", line 99, in on_read_data
RuntimeError: Unhandled Error
Traceback (most recent call last):
File "lib\multiprocessing\process.py", line 114, in run

File "TriblerGUI\core_manager.py", line 63, in start_tribler_core

File "lib\site-packages\twisted\internet\base.py", line 1195, in run

File "lib\site-packages\twisted\internet\base.py", line 1204, in mainLoop

--- <exception caught here> ---
File "lib\site-packages\twisted\internet\base.py", line 798, in runUntilCurrent

File "Tribler\Core\TFTP\handler.py", line 233, in data_came_in

File "Tribler\Core\TFTP\handler.py", line 325, in _handle_new_request

File "Tribler\Core\TFTP\handler.py", line 563, in _send_oack_packet

File "Tribler\Core\TFTP\handler.py", line 516, in _send_packet

File "Tribler\dispersy\candidate.py", line 30, in __init__

exceptions.AssertionError: ('182.30.65.219', 0)
```

--- a/Tribler/Core/TFTP/handler.py
+++ b/Tribler/Core/TFTP/handler.py
@@ -10,7 +10,8 @@ from twisted.internet import reactor
 
 from Tribler.dispersy.taskmanager import TaskManager, LoopingCall
 from Tribler.dispersy.candidate import Candidate
-from Tribler.dispersy.util import call_on_reactor_thread, blocking_call_on_reactor_thread, attach_runtime_statistics
+from Tribler.dispersy.util import (call_on_reactor_thread, blocking_call_on_reactor_thread, attach_runtime_statistics,
+                                   is_valid_address)
 from .session import Session, DEFAULT_BLOCK_SIZE, DEFAULT_TIMEOUT
 from .packet import (encode_packet, decode_packet, OPCODE_RRQ, OPCODE_WRQ, OPCODE_ACK, OPCODE_DATA, OPCODE_OACK,
                      OPCODE_ERROR, ERROR_DICT)
@@ -210,7 +211,7 @@ class TftpHandler(TaskManager):
         :param addr: The (IP, port) address tuple of the sender.
         :param data: The data received.
         """
-        if not self._is_running:
+        if not self._is_running or not is_valid_address(addr):
             return
 
         ip, port = addr

--- a/Tribler/Test/Core/TFTP/test_tftp_handler.py
+++ b/Tribler/Test/Core/TFTP/test_tftp_handler.py
@@ -76,6 +76,14 @@ class TestTFTPHandler(TriblerCoreTest):
         self.handler._is_running = False
         self.handler.data_came_in(None, None)
 
+    def test_data_came_in_invalid_candidate(self):
+        """
+        Testing whether we do nothing when data comes in from an invalid candidate
+        """
+        setattr(self.handler, "_process_packet",
+                lambda x, y: (_ for _ in ()).throw(RuntimeError("_process_packet may not be called")))
+        self.handler.data_came_in(('182.30.65.219', 0), None)
+
     def test_handle_new_request_no_metadata(self):
         """
         When the metadata_store from LaunchManyCore is not available, return

